### PR TITLE
bug fix - run_internal_graph()

### PR DESCRIPTION
**Pull request by farizrahman4u**
See #9565

### CODE REVIEW

File: keras/engine/topology.py [hunk 1]
```diff
@@ -2247,12 +2246,19 @@ def run_internal_graph(self, inputs, masks=None):
                             output_tensors = _to_list(layer.call(computed_tensors, **kwargs))
                             output_masks = _to_list(layer.compute_mask(computed_tensors,
                                                                        computed_masks))
-
                         # Apply activity regularizer if any:
                         if hasattr(layer, 'activity_regularizer') and layer.activity_regularizer is not None:
                             regularization_losses = [layer.activity_regularizer(x) for x in output_tensors]
                             layer.add_loss(regularization_losses, computed_tensors)
 
+                        if len(output_masks) == 1 and output_masks[0] is None and len(output_tensors) > 1:
```
[the remark this answers is not in the document]
Comment (farizrahman4u) [reply keras/engine/topology.py]: When a Lambda layer outputs multiple tensors but user doesn't explicitly provide mask (#9565).
Note that we can do this check within compute_mask() by calling the Lambda layer with the input and checking if it returns a tensor or list of tensors; but that would be expensive.

**Comment by farizrahman4u** [comment]
Code to reproduce error:

```python

from keras.layers import *
from keras.models import *


def f(x):
	return [x * 0.1, x * 0.2]


def output_shape(input_shape):
	return [input_shape, input_shape]


input = Input((10,))
x = Dense(10)(input)
x1, x2 = Lambda(f, output_shape=output_shape)(x)
output = multiply([x1, x2])
model = Model(input, output)  # No error here. predict() and fit() works.

input2 = Input((10,))
output2 = model(input2)  # >> error
model2 = Model(input2, output2)
```

File: keras/engine/topology.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fchollet) [reply keras/engine/topology.py]: I suggest to fix it in the following more-targeted way:

```
output_masks = layer.compute_mask(computed_tensors, computed_masks)
if output_masks is None:
   output_masks = [None for _ in output_tensors]
output_masks = _to_list(output_masks)
```

We should also add a unit test for this situation.

[the remark this answers is not in the document]
Comment (farizrahman4u) [reply keras/engine/topology.py]: Done.

**Comment by farizrahman4u** [comment]
@fchollet Done.

**Comment by mpariente** [comment]
Thanks a lot !

**Comment by jawadMansoor** [comment]
is this issue fixed, I am using Keras 2.2.4 , my lambda layer has list of outputs, I am unable to form this layer.... I followed this discussion https://github.com/keras-team/keras/pull/7222 was happy to read that it would be fixed in 2.0.8, so I upgraded keras but to no avail. any help? plz.

**Comment by farizrahman4u** [comment]
@jawadMansoor Please post a code snippet that reproduces the error.

**Comment by jawadMansoor** [comment]
x = Input(batch_shape=(batch_size, original_dim))
h = Dense(intermediate_dim[0], activation='relu')(x)
h = Dense(intermediate_dim[1], activation='relu')(h)
h = Dense(intermediate_dim[2], activation='relu')(h)
z_mean = Dense(latent_dim)(h)
z_log_var = Dense(latent_dim)(h)
z = Lambda(sampling, output_shape=(latent_dim,))([z_mean, z_log_var])
Gamma = Lambda(get_gamma,output_shape=[(batch_size,n_centroid),(n_centroid,),(n_centroid,latent_dim),(n_centroid,latent_dim)])(z)


unable to form last layer.

**Comment by jawadMansoor** [comment]
i was making a stupid mistake. sorry, list of outputs works fine. ty for your response though.

**Comment by KabaevAnton** [comment]
I think i have same issue reproduced in keras 2.2.4 
Here is my code:
`



def create_vae():
    models = {}
    #Encoder
    enc_input = Input(batch_shape=(batch_size, 28, 28, 1))
    X = Flatten()(enc_input)    
    X = Dense(256, activation='relu')(X)    
    X = Dense(128, activation='relu')(X)
    
    # Предсказываем параметры распределений
    # Вместо того, чтобы предсказывать стандартное отклонение, предсказываем логарифм дисперсии
    z_mean = Dense(latent_dim)(X)
    z_log_var = Dense(latent_dim)(X)
    
    # Сэмплирование из Q с трюком репараметризации
    def sampling(args):
        z_mean, z_log_var = args
        epsilon = K.random_normal(shape=(batch_size, latent_dim), mean=0., stddev=1.0)
        return z_mean + K.exp(z_log_var / 2) * epsilon
    
    l = Lambda(sampling, output_shape=(latent_dim,))([z_mean, z_log_var])
    
    models["z_meaner"] = Model(enc_input, z_mean, 'Enc_z_mean')
    models["z_lvarer"] = Model(enc_input, z_log_var, 'Enc_z_log_var')
    models["encoder"]  = Model(enc_input, l, 'Encoder') 
    
    
    # Декодер
    dec_input = Input(shape=(latent_dim, ))
    X = Dense(128)(dec_input)
    X = LeakyReLU()(X)    
    X = Dense(256)(X)
    X = LeakyReLU()(X)    
    X = Dense(28*28, activation='sigmoid')(X)
    decoded = Reshape((28, 28, 1))(X)
    
    models["decoder"] = Model(dec_input, decoded, name='Decoder')
    
    
    models["vae"]     = Model(enc_input, models["decoder"](models["encoder"](enc_input)), name="VAE")
    
    def vae_loss(x, decoded):
        x = K.reshape(x, shape=(batch_size, 28*28))
        decoded = K.reshape(decoded, shape=(batch_size, 28*28))
        xent_loss = 28*28*binary_crossentropy(x, decoded)
        kl_loss = -0.5 * K.sum(1 + z_log_var - K.square(z_mean) - K.exp(z_log_var), axis=-1)
        return (xent_loss + kl_loss)/2/28/28
    
    return models, vae_loss

models, vae_loss = create_vae()
`
